### PR TITLE
Add _endpoint_url argument to action methods

### PR DESCRIPTION
**Pull request by valaparthvi**
Resolves Issue #45.

### CODE REVIEW

**Comment by valaparthvi** [comment]
Not sure if unit tests are required for these changes. Let me know if they are.

**Comment by mkoura** [comment]
I like it, except of the name `_endpoint_url` (I know it was suggested in the discussion under https://github.com/ManageIQ/manageiq-api-client-python/issues/45). The underscore suggests internal use so as a programmer I would be hesitant to use it at all (not only as named argument) and would be tempted to pass `url=<endpoint_url>` instead.
I'd like something long that will likely not appear in payload (like `api_endpoint_url`) better.

@valaparthvi @cben your thoughts?

**Comment by valaparthvi** [comment]
Yeah, I was a little hesitant using that `_`, but we will probably never encounter an API field name starting with `_`, so an arg name starting with `_` is perhaps our best bet to avoid any future conflicts.
I like the name `api_endpoint_url` though.

**Comment by mkoura** [comment]
I like it :+1:
@cben can you review please?

**Comment by valaparthvi** [comment]
Hey @cben, can you review it and tell what you think?

**Comment by cben** [comment]
Sorry, had very busy week.  Reviewing...

File: src/manageiq_client/api.py [hunk 1]
```diff
@@ -105,23 +105,44 @@ def _sending_request(self, func, retries=2):
                 retries -= 1
         raise last_connection_exception
 
-    def get(self, url, **get_params):
-        self.logger.info("[RESTAPI] GET %s %r", url, get_params)
+    def get(self, api_endpoint_url=None, **get_params):
+        if not api_endpoint_url:
+            if "url" not in get_params:
+                raise ValueError(
+                    "'api_endpoint_url' or 'url' key must be specified in the parameters."
```
Comment (cben) [inline src/manageiq_client/api.py]: My only comment is consider not documenting a name at all, encourage passing as positional parameter.
(I reviewed various callers and all did positional anyway, except 1 place in integration_tests that I'll fix)
Something like:
```suggestion
                    "url must be specified, as positional paramater, or (deprecated) 'url' keyword parameter"
```
your call, LGTM either way (enough bikeshedding on my part ;-).

**Review by cben — APPROVED** [review]
:+1:

File: src/manageiq_client/api.py (hunk 1 above)
Comment (valaparthvi) [reply src/manageiq_client/api.py]: I like your way, I was skeptical about the message but didn't know what needed fix, this looks good.

**Pull request merged**